test/test_build.py

```diff
@@ -180,13 +217,20 @@ def test_image_is_generated(image_type):
 
 @pytest.mark.skipif(platform.system() != "Linux", reason="boot test only runs on linux right now")
 @pytest.mark.parametrize("image_type", gen_testcases("direct-boot"), indirect=["image_type"])
-def test_image_boots(image_type):
+def test_image_boots(image_type, build_ansible_container):
     with QEMU(image_type.img_path, arch=image_type.img_arch) as test_vm:
         exit_status, _ = test_vm.run("true", user=image_type.username, password=image_type.password)
         assert exit_status == 0
-        exit_status, output = test_vm.run("echo hello", user=image_type.username, password=image_type.password)
-        assert exit_status == 0
-        assert "hello" in output
+
+        distribution, version = get_distribution_and_version(test_vm, build_ansible_container,
+                                                             image_type.username, image_type.password)
+        assert distribution in ["Fedora", "CentOS"]
```

Comment:
I wonder if we should test in a more targeted way here, i.e. ensure that we get exactly the distro we want, right now that feels a little bit vague.

Reply:
yes, this is really bad - I just could not come up with a good example where I know all the parameters to check (other than the user check below) and the parameters of the image creation in this test just is "distro and user"